### PR TITLE
ConfirmModal: Migrates to React with new theme

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Updates implementation of ShowConfirmModalEvent to use react modal and upgrades to new Theme.

**Which issue(s) this PR fixes**:
Relates #32784

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx [hunk 1]
```diff
@@ -1,50 +1,86 @@
...
+  alternativeText,
   icon = 'exclamation-triangle',
   onConfirm,
   onDismiss,
-}) => {
-  const theme = useContext(ThemeContext);
-  const styles = getStyles(theme);
+  onAlternative,
+}: ConfirmModalProps): JSX.Element => {
+  const [disabled, setDisabled] = useState<boolean>(Boolean(confirmationText));
```
Comment (dprokop) [inline packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx:52]: I think `<boolean>` shouldn't be needed here

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx [hunk 2]
```diff
@@ -1,50 +1,86 @@
...
+  const styles = useStyles(getStyles);
+  const onConfirmationTextChange = (event: React.FormEvent<HTMLInputElement>) => {
+    setDisabled(confirmationText?.localeCompare(event.currentTarget.value, undefined, { sensitivity: 'accent' }) !== 0);
```
Comment (dprokop) [inline packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx:55]: Just wondering - do we need this localeCompare complexity here? Guess `===` would do it's job too :) Especially that typing "Delete" and "delete" makes it pass - don't know if this is a regression, probably not as we had `scope.confirmTextValid = payload.confirmText.toLowerCase() === value.toLowerCase();` before.

I think tho, that if the confitrm text is "Delete" then the input from the user should respect capitalization of D, as a matter of performing dangerous action

**Review by dprokop — APPROVED** [review]
Works well @hugohaggmark, awesome job! Left two nits to consider :)

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx (hunk 2 above)
Comment (hugohaggmark) [reply packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx:55]: I agree with your point that we should keep the exact capitalization. I think I'll change that. Regarding the `localeCompare` I think that is the most correct way of comparing texts according to this https://github.com/grafana/grafana/pull/32435#issuecomment-809934792 
https://stackoverflow.com/questions/2140627/how-to-do-case-insensitive-string-comparison/51005583#51005583

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx]: Fixed!

File: packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx (hunk 2 above)
Comment (hugohaggmark) [reply packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx]: Made it so we need the exact confirmation text instead, but still using localeCompare :)

**Pull request merged**